kedro/config/omegaconf_config.py

```diff
@@ -358,6 +345,43 @@ def load_and_merge_dir_config(
             if not k.startswith("_")
         }
 
+    @staticmethod
+    def _initialise_filesystem_and_protocol(
+        conf_source: str,
+    ) -> tuple[fsspec.AbstractFileSystem, str]:
+        """Set up the file system based on the file type detected in conf_source."""
+        file_mimetype, _ = mimetypes.guess_type(conf_source)
+        if file_mimetype == "application/x-tar":
+            protocol = "tar"
+        elif file_mimetype in (
+            "application/zip",
+            "application/x-zip-compressed",
+            "application/zip-compressed",
+        ):
+            protocol = "zip"
+        else:
+            protocol = "file"
+        fs = fsspec.filesystem(protocol=protocol, fo=conf_source)
+        return fs, protocol
+
+    def _merge_configs(
+        self,
+        config: dict[str, Any],
+        env_config: dict[str, Any],
+        key: str,
+        env_path: str,
+    ) -> Any:
+        merging_strategy = self.merge_strategy.get(key, "destructive")
```

Comment:
Shall we go even further and move merging strategies outside to the `Enum` and create a mapping of symbolic names to the implementations? Then all the possible values and implementations will be in one place, so it's easier to navigate and compare them, and additionally, we will get rid `if else` logic when applying a merging strategy.

Reply:
Some clarification on the above comment.

I was thinking of something like that:

```
class MergingStrategies(Enum):
    FIRST = "first"
    SECOND = "second"

def first_strat():
    pass
    
def second_strat():
    pass
    
merging_strategies_map = {
    MergingStrategies.FIRST: first_strat,
    MergingStrategies.SECOND: second_strat,
}
```

Usage:
```
first = strategies_map[MergingStrategies.FIRST]()
second = strategies_map[MergingStrategies("second")]()
```